src/UndisclosedPassword.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+
+namespace Zend\Validator;
+
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+
+final class UndisclosedPassword extends AbstractValidator
+{
+    const HIBP_API_URI = 'https://api.pwnedpasswords.com';
+    const HIBP_API_TIMEOUT = 300;
+    const HIBP_CLIENT_UA = 'zend-validator';
+    const HIBP_CLIENT_ACCEPT = 'application/vnd.haveibeenpwned.v2+json';
+    const HIBP_RANGE_LENGTH = 5;
+    const HIBP_RANGE_BASE = 0;
+    const HIBP_COUNT_BASE = 0;
+    const SHA1_LENGTH = 40;
+
+    const PASSWORD_BREACHED = 'passwordBreached';
+    const WRONG_INPUT = 'wrongInput';
+    const CONNECTION_FAILURE = 'connFail';
+    const UNKNOWN_ERROR = 'unknownError';
+
+    protected $messageTemplates = [
+        self::PASSWORD_BREACHED => 'The provided password was used by others',
+        self::WRONG_INPUT => 'The provided password failed to be correctly hashed, please verify your input',
+        self::CONNECTION_FAILURE => 'Unable to reach HIBP service, please try again later',
+        self::UNKNOWN_ERROR => 'Something happened beyond our control, error reporting should give more details',
+    ];
+
+    /**
+     * @var ClientInterface
+     */
+    private $httpClient;
+
+    /**
+     * @var RequestFactoryInterface
+     */
+    private $httpRequest;
+
+    /**
+     * @var ResponseFactoryInterface
+     */
+    private $httpResponse;
```

Comment:
s/httpResponse/makeHttpResponse maybe? (since this isn't a real response, but the factory for it)